examples/tensat.lisp

```diff
@@ -128,6 +128,20 @@
                  (when (and x y) (assert (equal x y)))
                  (values x nil)))))
 
+(define-analysis cost
+  :make (lambda (fsym &rest args)
+          (case fsym
+            ((ewadd ewmul) (let* ((shape (get-analysis-data (car args) 'shape)))
+              (reduce #'* shape)))
+            ((matmul)
+              (let* ((lhs (get-analysis-data (car args) 'shape))
+                    (rhs (get-analysis-data (car (last args)) 'shape)))
+                    (* (reduce #'* (append lhs (last rhs))))))
+            ;; TODO
+            ; ((conv2d) ...)
+            (otherwise (reduce #'max (mapcar (lambda (e) (get-analysis-data e 'cost)) args) :initial-value 0))))
+  :merge (lambda (x y) (min x y)))
```

Comment:
`merge` need to return two values, the second values should be `t` if analysis data has changed. You can copy the following from matmul.lisp:
```
(lambda (x y)
  (if (and (not x) y)
      (values y t)
    (values x nil)))
```

I should make common lattices into library functions at some point...